### PR TITLE
AirfRANS: Fourier + no-EMA on scarce + reynolds OOD tasks

### PR DESCRIPTION
**Pull request by tcapelle**
## Hypothesis

**Fourier features should improve OOD tasks as they improved the full task.** Fourier + no-EMA achieved -9.1% on the full task (#2457). Your no-EMA OOD baselines (#2460) showed scarce=0.2547, reynolds=0.4059 — both with pressure dominating >95% of error. Fourier encoding helps resolve high-frequency pressure gradients, which is exactly the bottleneck on OOD tasks.

## Current OOD Baselines (YOUR previous best!)

| Task | val_primary/surface_mse | PR |
|---|---|---|
| scarce | **0.2547** | #2460 (no-EMA, AdamW lr=5e-4, 2 epochs) |
| reynolds | **0.4059** | #2460 (no-EMA, AdamW lr=5e-4, 2 epochs) |

## Design

Two runs: Fourier + no-EMA on each OOD task:

| Run | Task | wandb_name |
|---|---|---|
| 1 | scarce | norman-fourier-noema-scarce |
| 2 | reynolds | norman-fourier-noema-reynolds |

## Instructions to Student

**CRITICAL: Run ONE AT A TIME. Never run parallel AirfRANS jobs.**

```bash
cd target/icml2026

# Run 1: Fourier + no-EMA on scarce task
python train.py --dataset airfrans --airfrans_task scarce \
  --optimizer adamw --lr 5e-4 --cosine_t_max 150 \
  --no-use-ema --enable-fourier \
  --slices 96 \
  --wandb_group norman-fourier-ood --wandb_name norman-fourier-noema-scarce

# Run 2: Fourier + no-EMA on reynolds task
python train.py --dataset airfrans --airfrans_task reynolds \
  --optimizer adamw --lr 5e-4 --cosine_t_max 150 \
  --no-use-ema --enable-fourier \
  --slices 96 \
  --wandb_group norman-fourier-ood --wandb_name norman-fourier-noema-reynolds
```

**Report** `val_primary/surface_mse`, per-channel MSE breakdown (especially surface_mse_p), test metrics, and epochs completed. Target: beat scarce=0.2547 and reynolds=0.4059.

### CODE REVIEW

**Comment by tcapelle** [comment]
STUDENT norman:

## Results

### Run 1: Fourier + no-EMA on scarce

| Metric | Baseline (#2460) | This run | Delta |
|---|---|---|---|
| val_primary/surface_mse | **0.2547** | 0.2760 | +8.4% ❌ |
| test_primary/surface_mse | — | 0.6892 | — |

**Val surface MSE breakdown (epoch 2):** Ux=0.000687, Uy=0.000540, nut=0.001634, p=1.1012

- **W&B run:** vb77cptv
- **Epochs:** 2 (30-min timeout)
- **Command:** `CUDA_VISIBLE_DEVICES=0 python train.py --dataset airfrans --airfrans-task scarce --optimizer adamw --lr 5e-4 --cosine-t-max 150 --no-use-ema --enable-fourier --agent norman --wandb-group norman-fourier-ood --wandb-name norman-fourier-noema-scarce`

### Run 2: Fourier + no-EMA on reynolds

| Metric | Baseline (#2460) | This run | Delta |
|---|---|---|---|
| val_primary/surface_mse | 0.4059 | **0.3319** | -18.2% ✅ |
| test_primary/surface_mse | — | 0.5881 | — |

**Val surface MSE breakdown (epoch 2):** Ux=0.000612, Uy=0.000067, nut=0.000080, p=1.3270

- **W&B run:** m24dt4cg
- **Epochs:** 2 (30-min timeout)
- **Command:** `CUDA_VISIBLE_DEVICES=0 python train.py --dataset airfrans --airfrans-task reynolds --optimizer adamw --lr 5e-4 --cosine-t-max 150 --no-use-ema --enable-fourier --agent norman --wandb-group norman-fourier-ood --wandb-name norman-fourier-noema-reynolds`

## Analysis

**Mixed results — Fourier helps reynolds significantly but hurts scarce.**

- **Reynolds: clear win.** Fourier + no-EMA drops val surface MSE from 0.4059 → 0.3319 (-18.2%). Pressure (p) still dominates >95% of error, but Fourier features helped the model resolve cross-Reynolds-number pressure gradients better.
- **Scarce: regression.** val surface MSE went from 0.2547 → 0.2760 (+8.4%). The scarce task has fewer training samples, so the extra Fourier feature dimensions may be adding noise rather than signal — classic overfitting-to-capacity tradeoff with limited data.
- **Both runs epoch-starved** at 2 epochs under the 30-min timeout. Both were still improving at cutoff (scarce: 0.3003→0.2760; reynolds: 0.4952→0.3319).
- **Pressure dominates** in both tasks (>95% of composite MSE), consistent with the full-task pattern.
- **Peak memory:** No explicit tracking in train.py; both runs completed without OOM on a single RTX PRO 6000 (98GB).

## Suggested follow-ups

1. **More epochs on reynolds** — still rapidly improving at epoch 2 cutoff, could push well below 0.33 with more training.
2. **Drop Fourier for scarce** — the scarce task may benefit more from regularization or data augmentation than from extra input features. The baseline no-EMA config (0.2547) remains the best approach for scarce.
3. **Fourier frequency tuning** — the default Fourier scales may not be optimal for OOD generalization; a sweep over frequency bands could help.

**Comment by tcapelle** [comment]
ADVISOR: Good work norman. Mixed but valuable results:
- **Reynolds: -18.2%** (0.3319 vs 0.4059) — Fourier clearly helps. New reynolds OOD baseline.
- **Scarce: +8.4%** (0.2760 vs 0.2547) — Fourier hurts scarce task, possibly overfitting with limited data.

Merging for the reynolds improvement. Scarce OOD baseline remains 0.2547 (no-Fourier, #2460).
